authentication_service/controllers/account_handlers.go

```
@@ -0,0 +1,160 @@
+}
+
+
+func SignUp() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var user models.User
+
+		if err := c.BindJSON(&user); err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+		validationErr := validate.Struct(user)
+		if validationErr != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": validationErr.Error()})
+			return
+		}
+		// validate whether the email is valid with edu
+		if !CheckEmailDomain(*user.Email, "edu") {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": "email is not valid"})
+			return
+		}
+
+		// check whether this email exist 
+		_, ok := database.UserCollection[*user.Email]
+		if ok {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": "email already exists"})
+			return
+		}
+		password := helper.HashPassword(*user.Password)
+		user.Password = &password
+
+		// create a new verification code for user
+		new_V_key := helper.EncodeToString(6)
+		user.V_key = &new_V_key
+
+		// save user in db
+		database.UserCollection[*user.Email] = user
+
+		// send an email
+		mail_service.SendVerificationCode(user, new_V_key)
+		c.JSON(http.StatusOK, gin.H{"Success": "Successful Sign Up"})
+	}
+}
+
+func Verify() gin.HandlerFunc {
+	return func(c *gin.Context){
+		var verification models.Verifiation
+
+		if err := c.BindJSON(&verification); err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+
+		user, ok := database.UserCollection[*verification.Email]
+		if !ok {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": "account does not exist"})
+			return
+		}
+
+		v_k := *user.V_key
+		// check whether verification code is correct
+		if v_k != *verification.V_key {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": "wrong validation code"})
+			return
+		}
+
+		// if verification all pass and correct then create access token
+		token, refreshToken, _ := helper.GenerateAllTokens(*user.Email, *user.First_name, *user.Last_name, *user.User_type)
```

idt we should ignore the error - currently, the `err` return on `GenerateAllTokens` is meaningless because we never do anything meaningful work with it. 

i feel like this is not a good pattern where we are crashing on actual errors, and ignoring any other errors that happen. would it be better to check the type of error returned and act appropriately?